apps/dashboard/app/javascript/packs/files/ClipBoard.js

```diff
@@ -0,0 +1,155 @@
+import ClipboardJS from 'clipboard'
+import Handlebars from 'handlebars';
+
+$(document).ready(function() {
+  
+  var clipBoard = new ClipBoard();
+
+  $("#directory-contents").on('success', function(e) {
+    $(e.trigger).tooltip({title: 'Copied path to clipboard!', trigger: 'manual', placement: 'bottom'}).tooltip('show');
+    setTimeout(() => $(e.trigger).tooltip('hide'), 2000);
+    e.clearSelection();
+  });
+  
+  $("#directory-contents").on('error', function(e) {
+    e.clearSelection();
+  });
+
+  $("#directory-contents").on("clipboardClear", function (e, options) {
+    clipBoard.clearClipboard();
+    clipBoard.updateViewForClipboard();
+  });
+
+  $("#directory-contents").on("updateClipboardFromSelection", function (e, options) {
+    clipBoard.updateClipboardFromSelection(options.selection);
+    clipBoard.updateViewForClipboard();
+  });
+
+  $("#directory-contents").on("updateViewForClipboard", function (e, options) {
+    clipBoard.updateViewForClipboard();
+  });
+
+
+});
+
+class ClipBoard {
+  _clipBoard = null;
+  _handleBars = null;
+
+  constructor() {
+    this._clipBoard = new ClipboardJS('#copy-path');
+    this._handleBars = Handlebars;

```

Comment:
Why do you need this? `Handlebars.compile` doesn't directly work?

Reply:
I can use Handlebars.compile, but if I'm going to create a class, then I should try not to rely on Global Variables.

Reply:
Changed HandleBars and Swal to be used as a globally scoped variable.